requests/auth.py

```diff
@@ -227,6 +227,11 @@ def handle_401(self, r, **kwargs):
         :rtype: requests.Response
         """
 
+        # If response is not 4xx, do not auth
```

Comment:
Given that I am clearly inclined to forget why we did this, let's add a link to the relevant issue here too. 😁